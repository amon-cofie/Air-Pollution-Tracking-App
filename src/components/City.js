import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router';
import { removeMagnify } from '../redux/cities';
import style from '../styles/City.module.css';

const City = (props) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const extras = useSelector((state) => state.extraData.allCountries);

  const { city, handleClick } = props;
  const stateName = city.state ? city.state : null;

  const fetchFlag = (str) => {
    let flag = '';
    extras.forEach((country) => {
      if (country.cca2 === str) {
        flag = country.flag;
      }
    });
    return flag;
  };

  return (
    <div
      className={style.city}
      onClick={() => {
        handleClick(city);
        navigate('/data');
        dispatch(removeMagnify());
      }}
    >
      <p className={style.cityName}>{city.name}</p>
      <p className={style.countryAlpha2}>
        {city.country}
        {' '}
        <span>{fetchFlag(city.country)}</span>
      </p>
      <p className={style.stateName}>{stateName}</p>
    </div>
  );
};

export default City;
