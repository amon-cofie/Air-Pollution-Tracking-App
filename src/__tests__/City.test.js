import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import City from '../components/City';

test('City component renders correctly', () => {
  const city = {};
  const snap = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <City city={city} />
      </BrowserRouter>
    </Provider>,
  );
  expect(snap).toMatchSnapshot();
});
