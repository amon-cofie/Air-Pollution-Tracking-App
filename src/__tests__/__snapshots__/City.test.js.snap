// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component renders correctly 1`] = `
<div
  className="city"
  onClick={[Function]}
>
  <p
    className="cityName"
  />
  <p
    className="countryAlpha2"
  >
     
    <span />
  </p>
  <p
    className="stateName"
  />
</div>
`;
